doc/SHOW_FORMATS_OPTION.md

```diff
@@ -0,0 +1,175 @@
+         }
+      ],
+      "ciphertext" : "$dynamic_0$aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+   }
+]
+```
+
+Or with one-liner in python:
+```
+$ john --show=formats a.pw | python -c 'import json, sys, pprint; pprint.pprint(json.loads(sys.stdin.read()))'
+[{u'ciphertext': u'$dynamic_0$aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+  u'lineNo': 1,
+  u'rowFormats': [{u'canonHash': [u'$dynamic_0$aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'],
+                   u'label': u'Raw-MD5',
+                   u'prepareEqCiphertext': True},
+                  {u'canonHash': [u'$dynamic_0$aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'],
+                   u'dynamic': True,
+                   u'label': u'dynamic_0',
+                   u'prepareEqCiphertext': True}]}]
+```
+
+There is a list of dictionaries with information for each line.
+
+A dictionary for line may contain such keys/fields:
+- "lineNo" is the number of line in file starting from 1,
+  - numbering is continuous among multiple files with hashes (it may be change in future versions),
+- "login" is for login,
+  - it may be absent if login is empty,
+  - it may be absent if login is not specified and line is skipped,
+  - it may contain dummy value "?" that john uses when login is not specified,
+- "ciphertext" is for ciphertext as it is extracted from hash file,
+  - it may be absent if ciphertext is empty (or cut by john to be empty),
+- "rowFormats" is a list for descriptions of john's formats that can load the line for cracking (see below),
+  - it may be empty list if line was skipped or none of formats can parse it,
+- "skipped" is to show that line is too short to be loaded by any format, so it is not passed to formats' checks at all,
+  - the value does not represent reason, the reason is always the same: the hash is too short and none of formats can load it,
+  - the value of this field is the origin of decision to skip, that's a label of branch in code that skipped the line (so you may check code in `loader.c`),
+- "uid", "gid", "gecos", "home", "shell" are for additional information about user (provided in some formats of hash files),
+  - they may be absent if they are empty,
+  - "gecos", "home", "shell" may be absent also if they have dummy value "/".
+
+"rowFormats" field contains a list of dictionaries with results of successful parsing of line by formats.
+
+Each dictionary in "rowFormats" list may have the following field/keys:
+- "label" is the name of format that may be used for `--format=` option,
+- "dynamic" is boolean value,
+  - it is true if format uses engine for dynamic formats,
+  - it is absent if it is false,
+- "prepareEqCiphertext" is boolean value,
+  - it is true if `prepare()` method of formats returned same ciphertext after processing (it may be interesting to developers of formats),
+  - it is absent if it is false,
+- "canonHash" is a list containing ciphertext in canonical form,
+  - it may contain multiple values for some formats of hash files (e.g. pwdump),
```

Comment:
It does not depend onto format of file. It is from format purely. 32 hex LM has 2 halves there.